Makefile

```diff
@@ -22,6 +22,9 @@ all: build/librubyparser.$(SOEXT)
 build/librubyparser.$(SOEXT): $(shell find src -name '*.c') $(shell find src -name '*.h') Makefile build include/yarp/ast.h
 	$(CC) $(OPTFLAGS) $(DEBUG_FLAGS) $(CFLAGS) -std=c99 -Wall -Werror -Wextra -Wpedantic -Wsign-conversion -fPIC -g -fvisibility=hidden -shared -Iinclude -o $@ $(shell find src -name '*.c')
 
+build/profile: $(shell find src -name '*.c') $(shell find src -name '*.h') Makefile build include/yarp/ast.h bin/profile.c
+	$(CC) $(CFLAGS) -O3 -std=c99 -Iinclude -o $@ $(shell find src -name '*.c') bin/profile.c
```

Comment:
Instead of recompiling all C files, one way would be to do something like `be rake clobber && be rake compile_no_debug && be rake compile`, or an easier rake command to get librubyparser compiled as `-O3` with no asserts.
And then link profile.c to that.

This line seems to not disable asserts (`-DNDEBUG=1`).

Reply:
Yes, but specifically the non-debug, `-O3` shared library.
Maybe we should store both libraries in different paths, that might be clearer.

In any case, feel free to merge this PR as-is, we can look at that later.